### DISCUSSION TITLE
first timers only issues?

### DISCUSSION THREAD

**Discussion by Genjidegaulle** [beetbox/beets#4855] [General]
I see in the documentation that there used to be a "first timers only" section for first time contributors -- but it doesn't seem like there are any now. 

Is there a reason no issues are categorized, or is it just that nothing's come up?

**Comment by sampsyo** [comment]
Hmm; I actually don't recall the exact intent behind creating that specific label… maybe the commit logs for `CONTRIBUTING.md` would reveal whose idea it was, who could then explain a little more?

We also have an older tag, "bitesize," which is similar in spirit and meant for listing self-contained units of work that would be good introductions to the contribution process:
https://github.com/beetbox/beets/labels/bitesize

**Comment by RollingStar** [comment]
Looks like there's some Github way to identify first issues. ("good first issue" tag)

https://docs.github.com/en/issues/using-labels-and-milestones-to-track-work/managing-labels

**We need to convert "bitesize" over to "good first issue" to use this feature**.

Issues with the good first issue label are used to populate the repository's contribute page. For an example of a contribute page, see [github/docs/contribute](https://github.com/github/docs/contribute).

https://github.com/beetbox/beets/contribute

https://github.blog/2020-01-22-browse-good-first-issues-to-start-contributing-to-open-source/

Found this on google, interesting: https://opensauced.pizza/blog/good-first-issues-dont-exist

> The best good first issues are the ones you open.

Organization owners can customize the default labels for repositories in their organization. For more information, see "[Managing default labels for repositories in your organization](https://docs.github.com/en/organizations/managing-organization-settings/managing-default-labels-for-repositories-in-your-organization)."

There are multiple "good first issue" search engines: https://goodfirstissue.dev/

And we can add beets to them (but should make it easier to install a dev version first, imo). See

https://discourse.beets.io/t/i-think-were-losing-most-potential-contributors-before-they-make-their-first-contribution/2148

https://discourse.beets.io/t/request-updated-dev-guide/1527

**Reply by sampsyo** [reply]
Great! I'm definitely down with just renaming the existing "bitesize" label to the standard name.

**Comment by RollingStar** [comment]
Bumping this,

We need to convert "bitesize" over to "good first issue" https://github.com/beetbox/beets/issues?q=is%3Aopen+is%3Aissue+label%3Abitesize

And could I maybe have label add permission (on all issues) if that doesn't require deeper permissions? Not sure how it works. I wonder if it would be good if everyone had label permissions on their own issues. Sometimes the maintainers use labels for triage / categorization, but as a first pass it might not be harmful for regular users to add that.

This I think is a "good first issue" - https://github.com/beetbox/beets/issues/5462

**Reply by JOJ0** [reply]
Thanks for bumping. Yes this important! Let's give motivated people a chance to find easy stuff quickly!

I'll see what I can do to give you the proper rights to fix those labels to _one_ name.

**Reply by RollingStar** [reply]
Thanks. Maybe I'm not looking hard enough, but

> **Organization owners** can manage default labels for repositories in the organization.

I'm showing as an "outside collaborator" which might be different. If you don't want to give that permission to me, I think if an "owner" creates the "good first issue" (all lowercase, with spaces) label, I should be able to assign existing bitesize issues to that new label.

https://docs.github.com/en/organizations/managing-organization-settings/managing-default-labels-for-repositories-in-your-organization

**Reply by JOJ0** [reply]
I assigned the triage permission which should allow you to add/remove existing labels. of all PRs and issues

I just realized that renaming existing labels probably is not possible with that permission so I just did the rename. You should be able to help assign/fix labels on existing issues and please help us do so! appreciated!

and sorry for the complications , we are currently unclear how permussions are managed and handled. organization owners permissions should be avoided.

**Reply by JOJ0** [reply]
should I delete the "first timers only" label? it's unused currently.

all that requires cleanup in the docs as well. you wanna open a pr for that @RollingStar?

**Reply by RollingStar** [reply]
Well it worked! 🎉 https://github.com/beetbox/beets/pull/5477 https://github.com/beetbox/beets/contribute 

> should I delete the "first timers only" label? it's unused currently.

Yes I think that's best.

> all that requires cleanup in the docs as well. you wanna open a pr for that @RollingStar?

This may be a 'good first issue' on its own. But no, I'm focusing on other beets stuff right now.
